detectable/src/main/java/com/blackduck/integration/detectable/detectables/gradle/inspection/parse/GradleReportTransformer.java

```diff
@@ -24,11 +24,127 @@
 public class GradleReportTransformer {
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
     private final EnumListFilter<GradleConfigurationType> configurationTypeFilter;
+    private final List<CodeLocation> allCodeLocationsWithinRoot = new ArrayList<>();
+    private final Map<String, CodeLocation> subProjectCodeLocationMap = new HashMap<>();
 
     public GradleReportTransformer(EnumListFilter<GradleConfigurationType> configurationTypeFilter) {
         this.configurationTypeFilter = configurationTypeFilter;
     }
 
+    public List<CodeLocation> transformRootReport(GradleReport gradleReport) {
```

Comment:
Have you considered trying to deduplicate some of the code between functions `transform/transformRootReport` and `addConfigurationToGraph/addConfigurationToRootAndSubProjectGraphs`?

Another thought is that this class is getting pretty complex and only half of it will be getting used on any given Detect run. Do you think it would be worth it to split it into two or more classes with common functionality extracted into a base class or similar? Then based on supplied user parameters we could instantiate a different set of classes in `DetectableFactory`.

Reply:
After trying out a few different refactoring paths, I went with the following wich I feel reduces repetition (and therefore improves maintainability compared to before the reafctoring) yet maintains readability. Let me know what you think:
- Kept both methods transform() and transformRootReportOnly() because the former returns a single code location, whereas the latter returns a list of code locations. 
- Extracted a chunk of common code between transform() and transformRootReportOnly() to a helper method called processConfigurations() 
- Extracted a chunk of common code that was repeated in 3 places to be inside an existing method called addGavToGraph()